src/test/java/com/basho/riak/client/core/util/DefaultCharsetTest.java

```diff
@@ -0,0 +1,152 @@
+package com.basho.riak.client.core.util;
+
+import org.junit.Assume;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.charset.Charset;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertEquals;
+
+public class DefaultCharsetTest
+{
+    @Test
+    public void testDefaultIfNoPropertyDefined()
+    {
+        Charset systemDefault = Charset.defaultCharset();
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean noCharsetPropertyDefined = definedCharsetName == null;
+        Assume.assumeTrue(noCharsetPropertyDefined);
+
+        assertEquals(systemDefault, clientDefault);
+    }
+
+    @Test
+    public void testDefinedCharset()
+    {
+        Charset clientDefault = DefaultCharset.get();
+
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+
+        Boolean charsetPropertyDefined = definedCharsetName != null && !definedCharsetName.isEmpty();
+
+        Assume.assumeTrue(charsetPropertyDefined);
+
+        Charset definedCharset = Charset.forName(definedCharsetName);
+        assertEquals(definedCharset, clientDefault);
+    }
+
+    @Test
+    public void testStaticInitializerValidDefinedCharset()
+            throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, NoSuchFieldException
+    {
+        String definedCharsetName = System.getProperty(Constants.CLIENT_OPTION_CHARSET);
+        Charset systemDefault = Charset.defaultCharset();
+        String goodCharsetName = "UTF-16";
+
+        try
+        {
+            Charset goodCharset = Charset.forName(goodCharsetName);
+
+            // Open up the private static initializer
+            Method staticInitializer = DefaultCharset.class.getDeclaredMethod("initializeDefaultCharsetSingleton",
+                                                                              null);
+            staticInitializer.setAccessible(true);
+
+            // Set system property for client charset to bad value
+            System.setProperty(Constants.CLIENT_OPTION_CHARSET, goodCharsetName);
+
+            // Run static initializer to test good charset handling
+            DefaultCharset initializedDefaultCharset = (DefaultCharset) staticInitializer.invoke(null);
+
+            // Get initialized charset
+            Field currentCharsetField = DefaultCharset.class.getDeclaredField("currentCharset");
```

Comment:
As we want to freeze the observable behavior, here is no needs to check internal state. better to verify as:

            assertEquals(systemDefault.name(), DefaultCharset.get().name());

Also it might be a good idea to use kind of a [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development) name for the test cases which are intended to freeze a  behavior.  Such approach will shed a bit more light on the expected behavior. For this case it might be something like:

    testTheDefaultSytemCharsetShouldBeUsedIfTheWrongOneWasRequested()

or something like this.